File: 969-number-of-recent-calls/number-of-recent-calls.py
class RecentCounter:

    def __init__(self):
        self.front = -1
        self.rear = -1
        self.queue = []

    def ping(self, t: int) -> int:
        if self.front == -1:
            self.front += 1
        self.queue.append(t)
        self.rear += 1
        count = 0

        for i in range(self.rear, self.front - 1, -1):
            if not (self.queue[self.rear] - 3000 <= self.queue[i] <= self.queue[self.rear]):
                break
            count += 1
        
        return count
        


# Your RecentCounter object will be instantiated and called as such:
# obj = RecentCounter()
# param_1 = obj.ping(t)

File: 969-number-of-recent-calls/number-of-recent-calls.java
class RecentCounter {
    int front,rear;
    ArrayList<Integer> ar;

    public RecentCounter() {
        this.front = -1;
        this.rear = -1;
        this.ar = new ArrayList<Integer>();
        
    }
    
    public int ping(int t) {
        if(front == -1){
            front++;
        }
        rear++;
        ar.add(t);
        
        int r = ar.get(rear) - 3000;
        while (ar.get(front) < r){
            front++;
        }

        return rear + 1 - front;
    }
}

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter obj = new RecentCounter();
 * int param_1 = obj.ping(t);
 */